### PR TITLE
Print Debugging Feature

### PR DESCRIPTION
**Pull request by georg-schwarz**
Adds a flag `dg` to the interpreter to determine the `debug granularity`. Value `exhaustive` and `peek` will debug-log the output of blogs. The former logs the complete output while the latter one will only show a subset of the data.

## TODO
- Investigate why the gtfs example stopped working and if it is connected with the changes in this PR.

### CODE REVIEW

File: libs/execution/src/lib/types/io-types/visitors/debug-log-visitor.ts [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+    if (this.debugGranularity !== 'peek') {
+      return;
+    }
+
+    this.log('... (omitted in peek mode)');
+  }
+
+  private log(text: string): void {
+    this.logger.logDebug(`[Result] ${text}`);
```
[the remark this answers is not in the document]
Comment (rhazn) [reply libs/execution/src/lib/types/io-types/visitors/debug-log-visitor.ts:139]: Or `Input` and `Output`.

**Review by rhazn — APPROVED** [review]
Neat, left some style comments. I'd only consider the naming ones essential (docs and debug output itself).

File: package.json [hunk 2]
```diff
@@ -60,6 +60,7 @@
     "@nx-plus/docusaurus": "^15.0.0-rc.0",
     "@types/jest": "28.1.1",
     "@types/mime-types": "^2.1.1",
+    "@types/multer": "^1.4.7",
```
[the remark this answers is not in the document]
Comment (georg-schwarz) [reply package.json:63]: My bad, seems like I switched terminal when I didn't intend to. Will remove the unnecessary dependency.

**Comment by georg-schwarz** [comment]
Thanks for the review @rhazn!
I incorporated all your comments.

Will investigate now on the failing gtfs example..

File: libs/execution/src/lib/types/io-types/visitors/debug-log-visitor.ts [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    if (this.debugGranularity === 'skip') {
+      return;
+    }
+
+    for (let i = 0; i < binaryFile.content.length; ++i) {
+      if (i > this.PEEK_NUMBER_OF_LINES) {
+        break;
+      }
+      this.log(`[Line ${i}] ${binaryFile.content[i] ?? '<undefined>'}`);
+    }
```
Comment (georg-schwarz) [inline libs/execution/src/lib/types/io-types/visitors/debug-log-visitor.ts:126]: Used `binaryFile.content.splice` before which turned out to operate directly on the data. Issue with the failing example is fixed by this @rhazn

**Pull request merged**